fix(common): Don't include statusText in HttpErrorResponse.message, deprecate HttpResponseBase.statusText

kescherCode opened the pull request:
Since HTTP/2, responses no longer contain a status text besides the status code, which caused our default value of 'OK' to be used in HttpErrorResponse.message. As discussed in #57580, we are opting to not change the default value to match the fetch spec (https://fetch.spec.whatwg.org/#concept-response-status-message), instead, we are deprecating the statusText field, and not using it in our own code.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #23334

## What is the new behavior?
The `HttpResponseBase.statusText` field is deprecated, and it is not used in `HttpErrorResponse.message` anymore.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information
This was discussed in #57580.

JeanMeche commented:
We'll need to update our golden files, for this please run `yarn bazel run //packages/common:common_api.accept`. 

Also please include a `DEPRECATED: ...` mention  in the footer of the commit message, as mentionned by [our guidelines](https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit-message-footer).

kescherCode committed:
fix(common): Don't include statusText in HttpErrorResponse.message, deprecate HttpResponseBase.statusText

Since HTTP/2, responses no longer contain a status text besides the status code, which caused our default value of 'OK' to be used in HttpErrorResponse.message.
As discussed in #57580, we are opting to not change the default value to match the fetch spec (https://fetch.spec.whatwg.org/#concept-response-status-message), instead, we are deprecating the statusText field, and not using it in our own code.

DEPRECATED: HttpResponseBase.statusText

This field is usually populated with OK by default, and overwritten by what a remote HTTP server responds with on the status line.
Since HTTP/2, we aren't guaranteed to have a value for this on the status line, meaning this value gets stuck at OK.
If you need this field for display purposes, it is recommended to retrieve a selection from RFC9110 (or future RFCs) yourself.
For other purposes, an HttpResponseCode enum is also provided at common/http.


Fixes #23334

gorvk commented:
any updates on this PR ?

kescherCode commented:
@gorvk not from my end. All requested changes have been made so far, so it has just been sitting here, waiting for further feedback, approval or a close.

thePunderWoman reviewed (approved):
LGTM

reviewed-for: fw-general, public-api

crisbeto reviewed (approved):
Reviewed-for: public-api
